spec/requests/notifications_spec.rb

```diff
@@ -1,24 +1,23 @@
 # frozen_string_literal: true
 
-RSpec.describe NotificationsController, type: :controller do
+describe 'Notifications', type: :request do
   describe '#destroy' do
-    let(:user) { FactoryBot.create(:user1) }
-    let(:other_user) { FactoryBot.create(:user2) }
+    let(:user) { create(:user1) }
+    let(:other_user) { create(:user2) }
     let(:notification_owner) { user }
     let!(:notification) do
-      FactoryBot.create(:notification, user: notification_owner)
+      create(:notification, user: notification_owner)
     end
 
     context 'when the user is signed in' do
       let(:previous_page) { 'http://example.com/previous' }
       let(:notification_id) { notification.id }
-      let(:format) { 'html' }
+      let(:headers) { { 'HTTP_REFERER' => previous_page } }
 
       before do
-        allow(controller).to receive(:current_user).and_return(user)
-        request.env['HTTP_REFERER'] = previous_page
+        sign_in user
 
-        delete :destroy, format: format, params: { id: notification_id }
+        delete notification_path(id: notification_id), headers: headers
       end
 
       context 'and the notification to be deleted exists' do
```

Comment:
These names should be refactored. "and" naming should not be used.
https://www.betterspecs.org/#contexts

Reply:
Changed `and the` to `when`  in context names.